lib/network.py

```diff
@@ -1933,6 +1939,23 @@ def get_proxies(self):
             return proxies
         return None
 
+    def on_bad_certificate(self, server, certificate):
+        if server in self.bad_certificate_servers:
+            return
+        self.bad_certificate_servers[server] = certificate
+        self.server_list_updated()
+
+    def remove_bad_certificate(self, server):
+        if not server in self.bad_certificate_servers:
```

Comment:
I remember reading some Python style guide that says `x not in d` is preferred to `not x in d`